malduck/disasm.py

```diff
@@ -114,20 +120,24 @@ def __eq__(self, other: Any) -> bool:
     def __str__(self) -> str:
         if self.is_imm:
             if self.x64:
-                return "0x%016x" % (self.value % 2 ** 64)
+                return "0x%016x" % (int(self.value) % 2 ** 64)
             else:
-                return "0x%08x" % (self.value % 2 ** 32)
-        if self.is_reg:
-            return self.reg
-        if self.is_mem:
+                return "0x%08x" % (int(self.value) % 2 ** 32)
+        elif self.is_reg:
+            return str(self.reg)
+        elif self.is_mem:
             s, m = [], self.mem
+            if m is None:
+                raise Exception("Invalid mem object")
             if m.base:
                 s.append(m.base)
             if m.index:
                 s.append("%d*%s" % (m.scale, m.index))
             if m.disp:
                 s.append("0x%08x" % (m.disp % 2 ** 32))
             return "%s [%s]" % (m.size, "+".join(s))
+        else:
+            raise Exception("Invalid Operand type")
```

Comment:
Maybe `"invalid operand"` would be better than raising Exceptions from `__str__`?

Reply:
Done